tests/integration/test_integration.py

```diff
@@ -752,3 +788,82 @@ async def test_given_vault_integrated_with_s3_when_restore_backup_then_action_fa
 
     backup_action_output = await run_restore_backup_action(ops_test, backup_id)
     assert backup_action_output.get("return-code") == 0
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.autounseal
+async def test_given_vault_is_deployed_when_integrate_another_vault_then_autounseal_activated(
```

Comment:
Does this cover the migration use case?

Reply:
No, I could add one for that, but that might start to get difficult considering we are brushing up against the limit the runners can handle. If we add any more units, we need to start doing cleanup.

There's also probably an argument that this doesn't need to be independently tested, because that implementation is all done on the Vault application side. We don't have any special logic to handle the migration other than setting the Charm status correspondingly.

Reply:
https://warthogs.atlassian.net/browse/TLSENG-297

Reply:
Yeah, that's a good point, it is all Vault logic.
Then it's good.